book/src/requirements.md

```diff
@@ -0,0 +1,102 @@
+
+**Rationale - UX:** Users can rely on the full privacy protections of Orchard even in the face of an Orchard soundness compromise, such as a successful countereiting attack.
+
+## Technical Strategy Requirements
+
+### TSR1. Sapling-equivalent Functionality
+
+**Requirement:** Unless other product or engineering concerns dictate, the functionality of
+Orchard should match that of the Sapling circuit.
+
+**Rationale - UX:** By being functional equivalent to Sapling, the existing Sapling
+userbase can immediately carry over most conceptual knowledge and UX habits from
+Sapling to Orchard.
+
+**Rationale - Technical Strategy:** Orchard is the first step in removing the toxic waste
+vulnerability as well as in deploying Halo 2. By targeting Sapling circuit functionality,
+the scope of Orchard R&D is greatly simplified to lower the strategic technical risk.
+
+### TSR2. Differences from Sapling Design Document
+
+**Requirement:** Every observable difference from Sapling from the perspective of a Mobile Wallet SDK, a zcashd RPC developer, or any user of any known services built with those components, must be documented in a "Differences from Sapling" document in this repository. Any change that impacts a product-level requirement must exist in this document as an explicit requirement.
+
+**Rationale - Technical Strategy:** Any deviation from Sapling needs to be clearly communicated to users and developers and reviewed by protocol engineers, product owners, and ecosystem stakeholders.
+
+### TSR3. Halo 2 Deployment
+
+**Requirement:** Orchard must use Halo 2 for zero knowledge proof validation.
+
+**Rationale - Technical Strategy:** By selecting Halo 2 as the ZKP platform for Orchard, the Orchard deployment provides a forcing function to productionize Halo 2.
+
+### TSR4. User Opt-in upon Activation
+
+**Requirement:** Users may individually shield funds or migrate into a Orchard Pool as soon as the feature activates.
+
+**Rationale - UX:** Users can act independently in opting into the Orchard Pool without coordinating globally.
+
+### TSR5. Full Design Verifiability
+
+**Requirement:** Any arbitrary future reviewer should be able to verify the entire design and public state of the Zcash blockchain without relying on external authorities. Note that the Trusted Setup ceremonies for Sprout and Sapling violate this goal, and while Orchard activation itself does not achieve the goal, it can be achieved if Sprout and Sapling pools are eventually retired.
+
+**Rationale - Technical Strategy:** A permissionless system requires the possibility of full-design validation by any arbitrary participant, and the need for this grows as the technology grows in importance.
+
+## Non-Requirements
+
+This section documents potential requirements that are explicitly not required, and their implementation is up to the product and engineering teams best judgement.
+
```

Comment:
```suggestion

### NonR1. Removing Sprout Functionality from the Consensus Protocol.

**Non-Requirement:** There will be no change in what is possible with Sprout funds at the consensus layer. (This leaves open the question of whether it will be possible to migrate funds directly from Sprout to Orchard within a transaction.)

**Rationale - Engineering:** This reduces the scope of consensus changes associated with NU5. Similar effects can be obtained by non-consensus restrictions (or just omission of functionality) in wallets. Consensus-level restrictions would have required changes in the migration design of [ZIP 308](https://zips.z.cash/zip-0308).

```